docs/custom_node_development.md

```diff
@@ -0,0 +1,125 @@
+# Custom Node Development Guide
+
+## Overview
+
+Custom node in OpenVINO Model Server simplifies linking deep learning models into a complete pipelines even if the inputs and output
+of the sequential models do not fit. In many cases, output of one model can not be directly passed to another one.
+The data might need to be analysed, filtered or converted to different format. Those operations can not be easily implemented
+in AI frameworks or are simply not supported. Custom node addresses this challenge. They allows employing a dynamic library
+developed in C++ or C to perform arbitrary data transformations. 
```

Comment:
While custom nodes were created with the purpose of simplifying the connections between models in DAG it can actually do more if one wishes since user can write basically any implementation there and just reuse our grpc/rest frontend (one cons is that such custom node execution is single threaded). 

@dtrawins / @dkalinowski I wonder if we want to mention it here as well?

Reply:
custom node execution could be multithreaded is someone decides to implement it that way. Anyway I think it is not the key use case to create standalone custom nodes without using OV and AI models at all.